File: VB/RangedMaskSample/MaskButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Mask
Imports DevExpress.Data.Mask
Imports System.Globalization

Namespace RangedMaskSample

	Public Class MaskButtonEdit
		Inherits ButtonEdit
		 Shared Sub New()
			RepositoryItemMaskButtonEdit.Register()
		 End Sub
		 Public Sub New()
			MaskElementsRepository.MaskButtonEditList.Add(Me)
		 End Sub

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemMaskButtonEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMaskButtonEdit)
			End Get
		End Property

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMaskButtonEdit.EditorName
			End Get
		End Property

		Public Sub CallLayoutChanged()
			Me.LayoutChanged()
		End Sub

		Protected Overrides Function CreateMaskManager(ByVal mask As MaskProperties) As MaskManager
			Dim patchedMask As New MaskButtonEditMaskProperties(Me)
			patchedMask.Assign(mask)
			patchedMask.EditMask = Properties.Mask.EditMask

			Return patchedMask.CreatePatchedMaskManager()
		End Function
	End Class
End Namespace

File: CS/RangedMaskSample/MaskButtonEdit.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Mask;
using DevExpress.Data.Mask;
using System.Globalization;

namespace RangedMaskSample
{

    public class MaskButtonEdit : ButtonEdit
    {
         static MaskButtonEdit()
        {             
            RepositoryItemMaskButtonEdit.Register();
        }
         public MaskButtonEdit()
        {
            MaskElementsRepository.MaskButtonEditList.Add(this);
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
         public new RepositoryItemMaskButtonEdit Properties
        {
            get { return base.Properties as RepositoryItemMaskButtonEdit; }
        }

        public override string EditorTypeName
        {
            get { return RepositoryItemMaskButtonEdit.EditorName; }
        }

        public void CallLayoutChanged()
        {
            this.LayoutChanged();
        }

        protected override MaskManager CreateMaskManager(MaskProperties mask)
        {
            MaskButtonEditMaskProperties patchedMask = new MaskButtonEditMaskProperties(this);
            patchedMask.Assign(mask);
            patchedMask.EditMask = Properties.Mask.EditMask;

            return patchedMask.CreatePatchedMaskManager();
        }
    }
}